internal/resources/data_source_dhcp_server.go

```diff
@@ -0,0 +1,57 @@
+// (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package resources
+
+import (
+	"context"
+
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/internal/utils"
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/pkg/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DhcpServerData() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: f(generalNamedesc, "DHCP Server", "DHCP Server"),
+			},
+			"id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "id can be obtained by using DHCP Server datasource/resource.",
+			},
+			"provider_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ProviderId of the DHCP Server. Use the provider_id  while creating DHCP NSX-T Segment Network",
+			},
+		},
+		ReadContext: DHCPServerReadContext,
+		Description: `The ` + DSDhcpServer + ` data source can be used to discover the ID of a hpegl vmaas dhcpserver.
```

Comment:
hpegl vmaas dhcpserver. Please DHCP (**All Caps**) in description fields

Reply:
done